crates/bevy_ecs/src/system/commands/command_queue.rs

```diff
@@ -0,0 +1,161 @@
+use super::Command;
+use crate::world::World;
+
+/// # Safety
+///
+/// This function is only every called when the `command` bytes is the associated
+/// [`Commands`] `T` type. Also this only reads the data via `read_unaligned` so unaligned
+/// accesses are safe.
+unsafe fn invoke_command<T: Command>(command: *mut u8, world: &mut World) {
+    let command = command.cast::<T>().read_unaligned();
+    command.write(world);
+}
+
+struct CommandMeta {
+    offset: usize,
+    func: unsafe fn(value: *mut u8, world: &mut World),
+}
+
+#[derive(Default)]
+pub(crate) struct CommandQueueInner {
+    bytes: Vec<u8>,
+    metas: Vec<CommandMeta>,
+}
+
+// SAFE: All commands [`Command`] implement [`Send`]
+unsafe impl Send for CommandQueueInner {}
+
+// SAFE: All commands [`Command`] implement [`Sync`]
```

Comment:
This is safe because `&CommandQueueInner` does not give access to the inner items (or any access at all)

This does not depend on `Command` being `Sync`